### PR TITLE
Cursorless tutorial ced

### PR DESCRIPTION
**Pull request by saidelike**
Currently requires https://github.com/saidelike/talon_hud/tree/cursorless fork of the talon HUD 

## Checklist

- [ ] Fix scope tree view
- [ ] Fix broken references to css eg reset
- [ ] Merge HUD changes
- [ ] Properly support custom actions and other necessary custom spoken forms
- [ ] Properly support custom symbol spoken forms (eg alphabet)
- [ ] Figure out how to properly detect VSCode focused
- [ ] Tune tutorial content? Maybe better for follow-up PR
- [ ] Highlight marks that need to be targeted, so user doesn't have to search for them
- [ ] Remove print statements and commented out code from `tutorial.py`
- [ ] Figure out how to only run step preparation when you have the right window focused. Some of the time @pokey gets error message after "cursorles help" step
   image
- [ ] Open new window on start?
- [ ] Ensure you're in the right window / editor before restoring state, so we don't accidentally clobber something
- [ ] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [ ] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [ ] I have not broken the cheatsheet

### CODE REVIEW

File: packages/cursorless-vscode/src/keyboard/grammar/generated/grammar.ts
Comment (pokey) [inline packages/cursorless-vscode/src/keyboard/grammar/generated/grammar.ts:1]: You shouldn't have to do this. Something is broken in your setup 🤔

File: packages/cursorless-vscode/src/registerCommands.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ import {
   TestCaseRecorder,
   showCheatsheet,
   updateDefaults,
+  tutorialCreate,
```
Comment (pokey) [inline packages/cursorless-vscode/src/registerCommands.ts:11]: I wouldn't directly import. Better to do what all the other commands do that come from engine: let the engine constructor in extension.ts pass a function / object out that gets passed into `registerCommands`. That way your function can use other things from our composition root if necessary, eg the hatTokenMap to force "air" token to be a particular token

**Review by pokey — COMMENTED** [review]
Left a few quick comments. Nice start!

**Comment by saidelike** [comment]
@pokey  I've pushed some very nice changes that make things being parsed from the extension side. Not everything is implemented but I would be curious if the general idea is fine with you.

**Comment by saidelike** [comment]
An initial version of the tutorial is live.
Please use https://github.com/saidelike/talon_hud/tree/cursorless fork/branch for the HUD

**Comment by pokey** [comment]
this PR seems to have pulled in some extraneous commits, so it's really hard to review. If you're not sure how to clean it up, lmk and I can take a swing

**Comment by saidelike** [comment]
> this PR seems to have pulled in some extraneous commits, so it's really hard to review 😅. If you're not sure how to clean it up, lmk and I can help you

It is because you had to merge the main branch to fix some issues iirc. You should be able to diff against that commit from the main branch maybe?

**Review by pokey — COMMENTED** [review]
Looking good! I made a [bunch of changes](https://github.com/cursorless-dev/cursorless/pull/2131/files/171abdd2e9e1d0580fb01bb6059041c7ba1b645d..0e1cf6bd83be5bc159682c57acc9a13eced8c4a8) and left some comments. I also added some checkboxes to the PR description

File: packages/cursorless-engine/src/core/TutorialImpl.ts [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+        `Can't file yaml file: ${yamlFile} in tutorial name: ${tutorialName}`,
+      );
+    }
+    const buffer = await readFile(yamlFile);
+    const fixture = yaml.load(buffer.toString()) as TestCaseFixture;
+
+    const editor = await ide().openUntitledTextDocument({
+      content: fixture.initialState.documentContents,
+      language: fixture.languageId,
+    });
```
[the remark this answers is not in the document]
Comment (saidelike) [reply packages/cursorless-engine/src/core/TutorialImpl.ts:192]: we have decided to move it to https://code.visualstudio.com/api/extension-guides/webview

**Comment by pokey** [comment]
closing in favour of the original https://github.com/cursorless-dev/cursorless/pull/360, as it's easier to use a branch that's part of the main repo

**Pull request closed without merging**